source/scripting_v3/GTA.Native/Native.cs

```diff
@@ -358,56 +360,254 @@ public T GetResult<T>()
+		public static T Call<T>(Hash hash, InputArgument argument0, InputArgument argument1)
+		{
 			unsafe
 			{
-				var res = SHVDN.NativeFunc.Invoke((ulong)hash, args);
+				var argPtr = stackalloc ulong[2];
 
-				// The result will be null when this method is called from a thread other than the main thread
-				if (res == null)
-				{
-					throw new InvalidOperationException("Native.Function.Call can only be called from the main thread.");
-				}
+				argPtr[0] = argument0._data;
+				argPtr[1] = argument1._data;
 
-				if (typeof(T).IsValueType || typeof(PoolObject).IsAssignableFrom(typeof(T)) || typeof(T).IsEnum)
-				{
-					return ObjectFromNative<T>(res);
-				}
-				else
-				{
-					return (T)ObjectFromNative(typeof(T), res);
-				}
+				return CallInternalNoParamArray<T>(hash, argPtr, 2);
+			}
+		}
+		/// <summary>
+		/// Calls the specified native script function and ignores its return value.
+		/// </summary>
+		/// <param name="hash">The hashed name of the script function.</param>
+		/// <param name="argument0">The first input or output argument to pass to the native script function.</param>
+		/// <param name="argument1">The second input or output argument to pass to the native script function.</param>
+		/// <param name="argument2">The third input or output argument to pass to the native script function.</param>
+		/// <returns>The return value of the native.</returns>
+		public static T Call<T>(Hash hash, InputArgument argument0, InputArgument argument1, InputArgument argument2)
+		{
+			unsafe
+			{
+				var argPtr = stackalloc ulong[3];
+
+				argPtr[0] = argument0._data;
+				argPtr[1] = argument1._data;
+				argPtr[2] = argument2._data;
+
+				return CallInternalNoParamArray<T>(hash, argPtr, 3);
+			}
+		}
+		/// <summary>
+		/// Calls the specified native script function and ignores its return value.
+		/// </summary>
+		/// <param name="hash">The hashed name of the script function.</param>
+		/// <param name="argument0">The first input or output argument to pass to the native script function.</param>
+		/// <param name="argument1">The second input or output argument to pass to the native script function.</param>
+		/// <param name="argument2">The third input or output argument to pass to the native script function.</param>
+		/// <param name="argument3">The fourth input or output argument to pass to the native script function.</param>
+		/// <returns>The return value of the native.</returns>
+		public static T Call<T>(Hash hash, InputArgument argument0, InputArgument argument1, InputArgument argument2, InputArgument argument3)
+		{
+			unsafe
+			{
+				var argPtr = stackalloc ulong[4];
+
+				argPtr[0] = argument0._data;
+				argPtr[1] = argument1._data;
+				argPtr[2] = argument2._data;
+				argPtr[3] = argument3._data;
+
+				return CallInternalNoParamArray<T>(hash, argPtr, 4);
+			}
+		}
+		static unsafe T CallInternalNoParamArray<T>(Hash hash, ulong* argPtr, int argCount)
```

Comment:
Not sure this function really helps readability. It's just two lines, so may as well duplicate those everywhere.